### ISSUE TITLE
errors for the file

### ISSUE THREAD

**Issue by dddfffggg777** [toptensoftware/yazd#14]
the "L727A:  NOP" added by the disassembler for some wrong reason
```
        ; --- START PROC L7278 ---
L7278:  LD      DE,0000h
        ; --- START PROC L727A ---
L727A:  NOP
        IM      1
        ; --- START PROC L727C ---
L727C:  LD      D,(HL)

```
[error.zip](https://github.com/toptensoftware/yazd/files/14379515/error.zip)

**Comment by dddfffggg777** [comment]
```
yazd v1.1.7.0 - Yet Another Z80 Disassembler
Copyright (C) 2012 Topten Software. All Rights Reserved.
Disassembler engine based on http://z80ex.sourceforge.net/

```

**Comment by toptensoftware** [comment]
Thanks for reporting.  I'll check it out, but I need the exact command line you're using to run the disassembler.

**Comment by dddfffggg777** [comment]
> Thanks for reporting. I'll check it out, but I need the exact command line you're using to run the disassembler.

included in ZIP
`yazd.exe --addr:29001 --entry:29001 swboot_cod swboot_cod.lst`

**Comment by toptensoftware** [comment]
OK, had a look at this.  The reason is because of this:

```
7257: 20 21                             JR      NZ,L727A
```

Because you've run the disassembler with `--entry` it traces the execution paths from all entry points - and any address that execution gets traced to gets disassembled from that point.

If you run with `--lst` arg as well you'll see this:

```
7278: 11 00 00                   L7278: LD      DE,0000h
                                        ; --- START PROC L727A ---
727A: 00                         L727A: NOP
727B: ED 56                             IM      1
```

ie: the `00` byte at address 0x727A is disassembled twice - once as `LD DE` instruction's immediate value, but it's also a jump target so also gets disassembled as a `NOP`.

If you run without the `--entry` switch the disassembler will just disassemble from top to bottom and things look a bit different.   Firstly the jump instruction has this note appended:

```
7257: 20 21                             JR      NZ,L7278+2      ; reference not aligned to instruction
```

and the rogue NOP isn't shown:

```
7278: 11 00 00                   L7278: LD      DE,0000h
727B: ED 56                      L727B: IM      1
```
In other words, the jump instruction at `0x7257` jumps to the middle of an instruction - so things get weird.

All up there's 70+ references not aligned to instructions - which is odd. Any chance this code is copy protected and trying to fool the disassembler?

**Comment by dddfffggg777** [comment]
here is calculated return addresses
like:
```
call print
db "this for print",0
call something_more


```

**Comment by toptensoftware** [comment]
ok, but I'm not sure what that has to do with this issue though.

**Comment by dddfffggg777** [comment]
> ok, but I'm not sure what that has to do with this issue though.

1. copy a code just cose it have reference inside is not ok
2. new parameter: file with address ranges treated as NOPs

**Comment by toptensoftware** [comment]
Sorry, don't understand.  Feel free to send a pull request if you want something changed and I'll consider it.

**Comment by dddfffggg777** [comment]
> Sorry, don't understand. Feel free to send a pull request if you want something changed and I'll consider it.

i can insert JPs to avoid code


```
call print
jr next
db "is for print",0
call something_more
next:
```

**Comment by toptensoftware** [comment]
Sorry, but I'm really not sure what you're asking for here.  

When using the tracing options (via the --entry command line arg) the behavior is to trace all execution paths and disassemble at those addresses.  If there are instructions that jump to the middle of other instructions (as per your original example) then the correct behavior is ambiguous, but the approach taken in yazd is for those overlapping bytes to be disassembled twice.  

That's just the way it's been implemented, and I don't see any reason to change it.

**Comment by dddfffggg777** [comment]
to force some areas into "db"

**Comment by toptensoftware** [comment]
Yes, anything that appears unreachable by tracing execution paths will be listed as data bytes.

**Comment by dddfffggg777** [comment]
> Yes, anything that appears unreachable by tracing execution paths will be listed as data bytes.

its not enough for the file

**Comment by toptensoftware** [comment]
I'm not sure what you're asking for.
